langchain_mcp_adapters/prompts.py

```
@@ -0,0 +1,52 @@
+from typing import Any, Optional, Union
+
+from langchain_core.messages import AIMessage, HumanMessage
+from mcp import ClientSession
+from mcp.types import PromptMessage
+
+
+class UnsupportedContentError(Exception):
+    """Raised when a prompt message contains unsupported content."""
+
+    pass
+
+
+def convert_mcp_prompt_message_to_langchain_message(
+    message: PromptMessage,
+) -> Union[HumanMessage, AIMessage]:
+    """Convert an MCP prompt message to a LangChain message.
+
+    Args:
+        message: MCP prompt message to convert
+
+    Returns:
+        a LangChain message
+    """
+    if message.content.type == "text":
+        if message.role == "user":
+            return HumanMessage(content=message.content.text)
+        if message.role == "assistant":
```

let's do `elif` and also raise if unsupported role